metricflow/constraints/time_constraint.py

```diff
@@ -17,6 +17,20 @@ class TimeRangeConstraint(SerializableDataclass):
     start_time: datetime.datetime
     end_time: datetime.datetime
 
+    def __post_init__(self) -> None:  # noqa: D
+        if self.start_time > self.end_time:
+            raise RuntimeError(
+                f"start_time must not be > end_time. start_time={self.start_time} end_time={self.end_time}"
+            )
+
+        if self.start_time < TimeRangeConstraint.ALL_TIME_BEGIN():
```

Comment:
Shall we switch this to epoch start instead of the year 2000? Seems more in line with other computer date systems.